### ISSUE TITLE
`no-unnecessary-type-assertion`: False positive on array-to-tuple assertion

### ISSUE THREAD

**Issue by ehoogeveen-medweb** [gajus/eslint-plugin-jsdoc#1756]
Typing a JavaScript array of literals with that type turns it into a tuple - a poor man's replacement for TypeScript's `as const`:
```js
const arr = ['foo'];
```
has type `string[]`,
```js
const arr = /** @type {['foo']} */ (['foo']);
```
has type `['foo']`.

While looking into this I learned that since TypeScript version 4.5, you can actually do this instead:
```js
const arr = /** @type {const} */ (['foo']);
```
which is much nicer. But the old way is not an unnecessary type assertion.

## Expected behavior

No warning *or* give a more specific warning and autofix to `/** @type {const} */`. 

## Actual behavior

The rule reports the type assertion as unnecessary and the autofix removes it, changing the type.

## ESLint Config

```js
import { jsdoc } from 'eslint-plugin-jsdoc';

export default [
  jsdoc({
    rules: {
      'jsdoc/no-unnecessary-type-assertion': 'error',
    },
  }),
];
```

## ESLint sample

```js
const arr = /** @type {['foo']} */ (['foo']);
```

## Environment

- Node version: 24.20.0
- ESLint version: 10.9.1
- `eslint-plugin-jsdoc` version: 64.3.3

**Comment by ehoogeveen-medweb** [comment]
Thanks for the quick fix!
